snapcraft/storeapi/v2/channel_map.py

```diff
@@ -0,0 +1,266 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2020 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Any, Dict, List, Optional, Set, Union
+
+"""
+This module holds representations for results for the v2 channel-map
+API endpoint provided by the Snap Store.
+
+The full API is documented on
+https://dashboard.snapcraft.io/docs/v2/en/snaps.html#snap-channel-map
+"""
+
+
+class Progressive:
+    """
+    Represents Progressive information for a MappedChannel.
+    """
+
+    @classmethod
+    def unmarshal(cls, payload: Dict[str, Union[str, bool]]) -> "Progressive":
+        return cls(
+            key=payload["key"],
```

Comment:
The payload data does needs to be validated on unmarshal to make sure we received the expected types.

In the meta, I've been doing something like:
```
key = payload.get("key")
if not isinstance(key, str):
  raise SomeError()
...
```

But it could be done more concisely with a schema check...

Reply:
done, with the schema, but I want it to raise on error to sentry as is if that is ok with you